test/model_test.dart

```diff
@@ -78,6 +78,29 @@ void main() {
     sdkAsPackageGraph = utils.testPackageGraphSdk;
   });
 
+  group('Tools', () {
+    Class toolUser;
+    Method invokeTool;
+    Method invokeToolNoInput;
+
+    setUp(() {
+      toolUser = exLibrary.classes.firstWhere((c) => c.name == 'ToolUser');
+      invokeTool =
+          toolUser.allInstanceMethods.firstWhere((m) => m.name == 'invokeTool');
+      invokeToolNoInput = toolUser.allInstanceMethods
+          .firstWhere((m) => m.name == 'invokeToolNoInput');
+      packageGraph.allLocalModelElements.forEach((m) => m.documentation);
+    });
+    test("can invoke a tool", () {
+      expect(invokeTool.documentation, contains('## `Yes it is!`'));
+    });
+    test(r"can invoke a tool with no $INPUT or args", () {
```

Comment:
Would like to see tests of cases where a comment reference is added/removed.

Reply:
OK, I added that, but I'm not sure how to verify that the link was actually recognized and linked.

Reply:
Ah.  You can use the member documentationAsHtml to do that.  https://github.com/dart-lang/dartdoc/blob/251962a97d3dd6566c104c7670017fdbc09e768e/lib/src/model.dart#L1163

There are some circumstances where this might not work at head; if that check doesn't work put a skip on the test and mention PR #1765.